.github/workflows/publish-package.yml

```diff
@@ -36,11 +45,25 @@ jobs:
       - name: Build
         run: yarn build
 
-      # TODO: we still need to make the tests work in CI environments
-      # will come back to this
-      # - name: Test
-      #   run: yarn test
+      - name: E2E tests
+        id: e2e
+        env:
+          API_KEY: ${{ secrets.API_KEY }}
+          OWNER_MNEMONIC: ${{ secrets.OWNER_MNEMONIC }}
+          PAYMASTER_POLICY_ID: ${{ secrets.PAYMASTER_POLICY_ID }}
+        run: yarn test:e2e
+
+      - name: Artifact of E2E tests
+        id: artifact
+        if: steps.e2e.outcome == 'success'
+        run: echo 'e2e=true' >> "$GITHUB_OUTPUT"
```

Comment:
from example in [docs](https://docs.github.com/en/actions/using-jobs/defining-outputs-for-jobs) on saving the "build and test" job's e2e step's output to condition whether to run the next job "deploy"